include/trieste/wf.h

```diff
@@ -262,9 +264,17 @@ namespace trieste
         return std::numeric_limits<size_t>::max();
       }
 
-      Sequence& operator[](size_t new_minlen)
+      Sequence& operator[](size_t new_len)
       {
-        minlen = new_minlen;
+        if (!has_minlen)
+        {
+          minlen = new_len;
+          has_minlen = true;
+        }
+        else
+        {
+          maxlen = new_len;
+        }
         return *this;
```

Comment:
Should we detect
```C++
Foo++[1][2][3]
```
as an error?  Would we error for maxlen < minlen?

Reply:
Neither of the errors give a strange behavior, but I guess it would be a slightly nicer user experience to check for it. I will add it...